src/memray/_memray/snapshot.cpp

```diff
@@ -19,6 +20,18 @@ HighWaterMarkLocationKey::operator==(const HighWaterMarkLocationKey& rhs) const
            && native_segment_generation == rhs.native_segment_generation && allocator == rhs.allocator;
 }
 
+bool
+operator<(const AllocationLifetime& lhs, const AllocationLifetime& rhs)
+{
+    // Sort by n_bytes if allocatedBefore/deallocatedBefore are equal,
+    // so that our test suite gets records in a predictable order.
+    return lhs.allocatedBeforeSnapshot != rhs.allocatedBeforeSnapshot
+                   ? lhs.allocatedBeforeSnapshot < rhs.allocatedBeforeSnapshot
+           : lhs.deallocatedBeforeSnapshot != rhs.deallocatedBeforeSnapshot
+                   ? lhs.deallocatedBeforeSnapshot < rhs.deallocatedBeforeSnapshot
+                   : lhs.n_bytes < rhs.n_bytes;
+}
```

Comment:
I find the chain of ternary stuff a bit difficult to read. What about something like:

```
bool operator<(const AllocationLifetime& lhs, const AllocationLifetime& rhs) {
  if (lhs.allocatedBeforeSnapshot != rhs.allocatedBeforeSnapshot) {
    return lhs.allocatedBeforeSnapshot < rhs.allocatedBeforeSnapshot;
  } else if (lhs.deallocatedBeforeSnapshot != rhs.deallocatedBeforeSnapshot) {
    return lhs.deallocatedBeforeSnapshot < rhs.deallocatedBeforeSnapshot;
  } else {
    return std::tie(lhs.n_bytes) < std::tie(rhs.n_bytes);
  }
}
```